sequencers/based/sequencer.go

```diff
@@ -0,0 +1,172 @@
+package based
```

Comment:
the scope of this pr seems to be creeping away from blockmanager only. can we split  it up